distribution/src/bin/elasticsearch

```diff
@@ -45,6 +47,22 @@ then
   fi
 fi
 
+if [[ $ATTEMPT_SECURITY_AUTO_CONFIG = true ]]; then
+  # It is possible that an auto-conf failure prevents the node from starting, but this is only the exceptional case (exit code 1).
+  # Most likely an auto-conf failure will leave the configuration untouched (exit codes 73, 78 and 80), optionally printing a message
+  # if the error is uncommon or unexpected, but it should otherwise let the node to start as usual.
+  # It is passed in all the command line options in order to read the node settings ones (-E), while the other parameters are ignored
+  # (a small caveat is that it also inspects the -v option in order to provide more information on how auto config went)
+  if ! bin/elasticsearch-security-config "$@" <<<"$KEYSTORE_PASSWORD"; then
```

Comment:
I think we need to have another round of discussion at some point on whether we should be even handling password protected `elasticsearch.keystore`s. Given where this runs in the startup process, for the password to the keystore to be set, it means that either the elasticsearch.keystore pre-existed or the user has password protected it themselves before starting the node in archives. Arguably, this can be considered an _explicit_ security related configuration and we can alternatively bail.

Reply:
If you suggest we remove the tool, I have nothing against it.

Reply:
There are non-security settings in the keystore, be it password protected or not.
I don't think branching the logic on the condition of keystore password protection is useful.

Reply:
>  I don't think branching the logic on the condition of keystore password protection is useful.

I'm happy to defer this discussion at a later point if any and I don't have a strong opinion either. 

> If you suggest we remove the tool, I have nothing against it.

Yes this is my suggestion